### PR TITLE
async query status change 

### PR DESCRIPTION
**Pull request by sumukhswamy**
### Description
Fixed async polling with cancel
added backticks
fixed sidebar issue of loading spinner when datasource changed

 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/components/Main/main.tsx [hunk 1]
```diff
@@ -528,7 +528,7 @@ export class Main extends React.Component<MainProps, MainState> {
         response as IHttpResponse<ResponseData>
       );
       const status = result.data['status'];
```
Comment (ps48) [inline public/components/Main/main.tsx:530]: Should also add a conversion to lower `status.toLowerCase()`

**Review by paulstn — COMMENTED** [review]
https://github.com/opensearch-project/dashboards-query-workbench/blob/main/public/components/QueryResults/async_query_body.tsx#L54
async query body looks for a "FAILED" string, you need to change this too

**Comment by sumukhswamy** [comment]
@paulstn please check this again, I have changed the QueryResults test as well, please let me know if I have missed anything else?

**Review by paulstn — APPROVED** [review]
LGTM!

**Pull request closed without merging**